flytekit/bin/entrypoint.py

```diff
@@ -222,98 +224,141 @@ def _handle_annotated_task(task_def: PythonTask, inputs: str, output_prefix: str
+        if not isinstance(_task_def, PythonFunctionTask):
+            raise Exception("Cannot be run with instance tasks.")
+        map_task = MapPythonTask(_task_def, max_concurrency)
 
         task_index = _compute_array_job_index()
         output_prefix = _os.path.join(output_prefix, str(task_index))
 
-        _handle_annotated_task(map_task, inputs, output_prefix, raw_output_data_prefix)
+        if test:
+            _click.echo(
+                f"Test detected, returning. Inputs: {inputs} Computed task index: {task_index} "
+                f"New output prefix: {output_prefix} Raw output path: {raw_output_data_prefix} "
+                f"Resolver and args: {resolver} {resolver_args}"
+            )
+            return
 
-    # Old style task
-    elif not test:
-        raise _system_exceptions.FlyteSystemAssertion(
-            "Map tasks are only supported for task of type `PythonTask` created using flytekit>=0.16.0"
-            " got task {} of type {} instead".format(task_name, type(task_def))
-        )
+        _handle_annotated_task(map_task, inputs, output_prefix, raw_output_data_prefix)
 
 
 @_click.group()
 def _pass_through():
     pass
 
 
-_task_module_option = _click.option("--task-module", required=True)
-_task_name_option = _click.option("--task-name", required=True)
-_inputs_option = _click.option("--inputs", required=True)
-_output_prefix_option = _click.option("--output-prefix", required=True)
-_raw_output_date_prefix_option = _click.option("--raw-output-data-prefix", required=False)
-_test = _click.option("--test", is_flag=True)
-
-
 @_pass_through.command("pyflyte-execute")
-@_click.option("--task-module", required=True)
-@_click.option("--task-name", required=True)
+@_click.option("--task-module", required=False)
+@_click.option("--task-name", required=False)
 @_click.option("--inputs", required=True)
 @_click.option("--output-prefix", required=True)
 @_click.option("--raw-output-data-prefix", required=False)
 @_click.option("--test", is_flag=True)
-def execute_task_cmd(task_module, task_name, inputs, output_prefix, raw_output_data_prefix, test):
+@_click.option("--resolver", required=False)
+@_click.argument(
+    "resolver-args", type=_click.UNPROCESSED, nargs=-1,
+)
+def execute_task_cmd(
+    task_module, task_name, inputs, output_prefix, raw_output_data_prefix, test, resolver, resolver_args
+):
     _click.echo(_utils.get_version_message())
     # Backwards compatibility - if Propeller hasn't filled this in, then it'll come through here as the original
     # template string, so let's explicitly set it to None so that the downstream functions will know to fall back
     # to the original shard formatter/prefix config.
     if raw_output_data_prefix == "{{.rawOutputDataPrefix}}":
         raw_output_data_prefix = None
 
-    _execute_task(task_module, task_name, inputs, output_prefix, raw_output_data_prefix, test)
+    # For new API tasks, we need to call a different function.
```

Comment:
include which version of the API(so new contributors can understand what's going on)?